core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts

```diff
@@ -759,6 +771,49 @@ export class BaseUser {
     this.page = newPage;
     return newPage;
   }
+
+  /**
+   * This function sets up video recording for the current test.
+   */
+  async setupVideoRecording(): Promise<void> {
+    const outputFileName =
+      `${this.specName}-${new Date().toISOString()}.mp4`.replace(
+        /[^a-z0-9.-]/gi,
+        '_'
+      );
+
+    const outputDir = testConstants.TEST_VIDEO_DIR;
+    if (!fs.existsSync(outputDir)) {
+      fs.mkdirSync(outputDir, {recursive: true});
+    }
+
+    // Skip if there is no page object.
```

Comment:
Can you explain why -- i.e. in which situations is there "no page object"?

Reply:
While we always run `setupVideoRecording` method when there is a page object. However, it's not true if called before `openBrowser` method. So, it's proofing against someone using this method before `openBrowser` method.

Considering this and previous comment, should I implement Video recording in `openBrowser` method itself?

Reply:
Sorry, I couldn't understand. Should I move recording logic to `openBrowser` method instead of it's explicit function?

Reply:
I think you can do that, yeah.

Reply:
This check is no longer required, as we are implementing screen recording in openBrowser itself, once page object is created.